src/models/Portfolio.ts

```diff
@@ -1,12 +1,40 @@
 import { IUser } from './User'
-import { Schema, model, Model, ObjectId } from 'mongoose'
+import { Schema, model, Model, ObjectId, Date } from 'mongoose'
+
+type Status =
+  | 'portfolio_not_submitted'
+  | 'under_review'
+  | 'portfolio_needs_revision'
+  | 'portfolio_revision_exceeded'
+  | 'portfolio_passed_interview_to_be_scheduled'
 
 export interface IPortfolioUrl {
+  /** Current submission number for the portfolio, number*/
   submissionNo: number
-  resubmissionNo?: number
+  /** This array stores the previous numbers of the submitted portfolio, Array<number>*/
+  previousSubmissions?: Array<{
+    number: number
+    url: string
+    date: Date
+  }>
+  /** This stores the times the portfolio has been resubmitted, number*/
+  resubmissionCount?: number
+  /** This stores the current URL of the portfolio*/
   portfolioUrl: string
-  status: string
-  reviewComment?: string | undefined
+  /** This stores the current status of the portfolio */
+  status: Status
+  reviewComments?: Array<{
+    author: string
+    comment: string
+    date: Date
+  }>
+  portfolioMarks?: {
+    projects: number
+    blogs: number
+    effort: number
+    linkedin: number
+  }
+  // TODO: Come up with better type
   // eslint-disable-next-line @typescript-eslint/no-explicit-any
   user: any
```

Comment:
Here, I guess instead of `any` we can keep it as `IUser`, because we won't send just an `id` to frontend, we will populate it which means it will become as object as `user` model

```suggestion
  user: IUser
```